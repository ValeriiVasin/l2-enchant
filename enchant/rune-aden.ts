import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const runeAdenEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.RuneAden_1,
  chances: [70, 60, 50, 40, 60, 55, 50, 45, 40, 50, 45, 40, 35, 30, 30, 25, 20, 15, 10],
  items: [
    Item.RuneAden_1,
    Item.RuneAden_2,
    Item.RuneAden_3,
    Item.RuneAden_4,
    Item.RuneAden_5,
    Item.RuneAden_6,
    Item.RuneAden_7,
    Item.RuneAden_8,
    Item.RuneAden_9,
    Item.RuneAden_10,
    Item.RuneAden_11,
    Item.RuneAden_12,
    Item.RuneAden_13,
    Item.RuneAden_14,
    Item.RuneAden_15,
    Item.RuneAden_16,
    Item.RuneAden_17,
    Item.RuneAden_18,
    Item.RuneAden_19,
    Item.RuneAden_20,
  ],
  boundaries: new Set([Item.RuneAden_6, Item.RuneAden_11, Item.RuneAden_16]),
});
