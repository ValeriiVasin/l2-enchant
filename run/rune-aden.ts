import { runeAdenEnchantMap } from '../enchant/rune-aden';
import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.RuneAden_1, itemsRange(Item.RuneAden_2, Item.RuneAden_20), 100_000, runeAdenEnchantMap);
