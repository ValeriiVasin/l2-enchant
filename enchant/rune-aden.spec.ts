import { runeAdenEnchantMap } from './rune-aden';

it('enchant map: rune aden', () => {
  expect(runeAdenEnchantMap).toMatchSnapshot();
});
