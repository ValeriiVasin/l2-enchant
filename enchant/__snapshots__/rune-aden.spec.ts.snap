// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: rune aden 1`] = `
Map {
  "Кристалл Души Адена Ур.1" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.1",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Кристалл Души Адена Ур.2",
      },
      Object {
        "chance": 30,
        "item": "Кристалл Души Адена Ур.1",
      },
    ],
  },
  "Кристалл Души Адена Ур.2" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.2",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Кристалл Души Адена Ур.3",
      },
      Object {
        "chance": 40,
        "item": "Кристалл Души Адена Ур.1",
      },
    ],
  },
  "Кристалл Души Адена Ур.3" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.3",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Кристалл Души Адена Ур.4",
      },
      Object {
        "chance": 50,
        "item": "Кристалл Души Адена Ур.1",
      },
    ],
  },
  "Кристалл Души Адена Ур.4" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.4",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Кристалл Души Адена Ур.5",
      },
      Object {
        "chance": 60,
        "item": "Кристалл Души Адена Ур.1",
      },
    ],
  },
  "Кристалл Души Адена Ур.5" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.5",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Кристалл Души Адена Ур.6",
      },
      Object {
        "chance": 40,
        "item": "Кристалл Души Адена Ур.1",
      },
    ],
  },
  "Кристалл Души Адена Ур.6" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.6",
    "results": Array [
      Object {
        "chance": 55,
        "item": "Кристалл Души Адена Ур.7",
      },
      Object {
        "chance": 45,
        "item": "Кристалл Души Адена Ур.6",
      },
    ],
  },
  "Кристалл Души Адена Ур.7" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.7",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Кристалл Души Адена Ур.8",
      },
      Object {
        "chance": 50,
        "item": "Кристалл Души Адена Ур.6",
      },
    ],
  },
  "Кристалл Души Адена Ур.8" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.8",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Кристалл Души Адена Ур.9",
      },
      Object {
        "chance": 55,
        "item": "Кристалл Души Адена Ур.6",
      },
    ],
  },
  "Кристалл Души Адена Ур.9" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.9",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Кристалл Души Адена Ур.10",
      },
      Object {
        "chance": 60,
        "item": "Кристалл Души Адена Ур.6",
      },
    ],
  },
  "Кристалл Души Адена Ур.10" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.10",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Кристалл Души Адена Ур.11",
      },
      Object {
        "chance": 50,
        "item": "Кристалл Души Адена Ур.6",
      },
    ],
  },
  "Кристалл Души Адена Ур.11" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.11",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Кристалл Души Адена Ур.12",
      },
      Object {
        "chance": 55,
        "item": "Кристалл Души Адена Ур.11",
      },
    ],
  },
  "Кристалл Души Адена Ур.12" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.12",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Кристалл Души Адена Ур.13",
      },
      Object {
        "chance": 60,
        "item": "Кристалл Души Адена Ур.11",
      },
    ],
  },
  "Кристалл Души Адена Ур.13" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.13",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Кристалл Души Адена Ур.14",
      },
      Object {
        "chance": 65,
        "item": "Кристалл Души Адена Ур.11",
      },
    ],
  },
  "Кристалл Души Адена Ур.14" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.14",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Кристалл Души Адена Ур.15",
      },
      Object {
        "chance": 70,
        "item": "Кристалл Души Адена Ур.11",
      },
    ],
  },
  "Кристалл Души Адена Ур.15" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.15",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Кристалл Души Адена Ур.16",
      },
      Object {
        "chance": 70,
        "item": "Кристалл Души Адена Ур.11",
      },
    ],
  },
  "Кристалл Души Адена Ур.16" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.16",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Кристалл Души Адена Ур.17",
      },
      Object {
        "chance": 75,
        "item": "Кристалл Души Адена Ур.16",
      },
    ],
  },
  "Кристалл Души Адена Ур.17" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.17",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Кристалл Души Адена Ур.18",
      },
      Object {
        "chance": 80,
        "item": "Кристалл Души Адена Ур.16",
      },
    ],
  },
  "Кристалл Души Адена Ур.18" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.18",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Кристалл Души Адена Ур.19",
      },
      Object {
        "chance": 85,
        "item": "Кристалл Души Адена Ур.16",
      },
    ],
  },
  "Кристалл Души Адена Ур.19" => Object {
    "cost": "Кристалл Души Адена Ур.1",
    "item": "Кристалл Души Адена Ур.19",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Кристалл Души Адена Ур.20",
      },
      Object {
        "chance": 90,
        "item": "Кристалл Души Адена Ур.16",
      },
    ],
  },
}
`;
